protocol/contracts/governance/ElderCandidate.sol

```diff
@@ -0,0 +1,163 @@
+//SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
+
+/**
+ * @title an NFT for each elder candidate, with associated endorsements
+ * 
+ * @notice Captures key aspects of temple governance on chain. Namely
+ * 
+ *   - Anyone in temple discord with at least the Templar Role + verified
+ *     wallet can nominate themselves for candidacy.
+ *   - Any user can endorse candidates
+ *   - EIP712 structured data signing to allow 'gasless voting' via a relay
+ * 
+ * Which elders are currently voted in and responsible for approving proposals is handled
+ * at the social layer.
+ */
+contract ElderCandidate is ERC721, Ownable, EIP712 {
+    using Strings for uint256;
+    using Counters for Counters.Counter;
+
+    bytes32 public immutable TOGGLE_ENDORSEMENTS_FOR_TYPEHASH = keccak256("toggleEndorsementsFor(address account, uint256[] discordIds, uint256 deadline, uint256 nonce)");
+    mapping(address => Counters.Counter) public _nonces;
+
+    /// @notice endorsements per nomination
+    mapping(address => mapping (uint256 => bool)) public endorsementsBy;
+
+    /// @dev merkle tree root of (address, claimShare)
+    bytes32 public merkleRoot;
+
+    /// @notice Token which nomination fee is charged in
+    IERC20 public immutable nominationFeeToken;
+
+    /// @notice Nomination fee amount (in feeToken)
+    uint256 public nominationFee;
+
+    constructor(
+        IERC20 _nominationFeeToken,
+        uint256 _nominationFee
+    ) ERC721("Temple Elder Candidate", "ELDER CANDIDATE") EIP712(name(), "1") {
+        nominationFeeToken = _nominationFeeToken;
+        nominationFee = _nominationFee;
+    }
+
+    /**
+     * @dev Change the nomination fee
+     * 
+     * Designed to be low, but a defense in depth to stop peeps putting up a nomination
+     * for the lolz.
+     */
+    function setNominationFee(uint256 _nominationFee) external onlyOwner {
+        nominationFee = _nominationFee;
```

Comment:
> Slightly concerned this will dissuade people from voting when they otherwise might - ie a negative effect?

I think  this fee is about self nominating to be an elder, not voting?